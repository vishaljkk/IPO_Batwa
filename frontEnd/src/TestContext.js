import React, { useContext } from "react";

const initialState = {
    bankbalance:100000,
    bids: [],
    error: null,
    loading: true
}  

const GlobalContext = React.createContext(initialState);
export default GlobalContext;
export const useMyContext = () => useContext(GlobalContext);
