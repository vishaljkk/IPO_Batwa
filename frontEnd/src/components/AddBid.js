import React, {useState, useContext} from 'react'
import { GlobalContext } from '../context/GlobalState';

export const AddBid = () => {
    const { addTransaction } = useContext(GlobalContext);
    const onSubmit = e =>{
        e.preventDefault();
        const newBid = {
            text,
            amount: +amount*quantity
        }
        addTransaction(newBid);
    }
    const [text, setText] = useState('');
    const [amount, setAmount] = useState('');
    const [quantity, setQuantity] = useState(1);
    return (
        <div>
            <h3>Add Bid</h3>
            <form onSubmit={onSubmit}>
                <div className="form-control">
                <label htmlFor="text">Select IPO name</label>
                <input type="text" value={text}  onChange={(e) => setText(e.target.value)} placeholder="Enter IPO name..." />
                </div>
                <div className="form-control">
                <label htmlFor="amount">
                    Enter Number of Lots <br />
                </label>
                <input type="number" value={quantity} onChange={(e) => setQuantity(e.target.value)} placeholder="Enter number of lots..." />
                <label htmlFor="amount"
                    >Price Of one lot <br />
                </label>
                <input type="number" value={amount} onChange={(e) => setAmount(e.target.value)} placeholder="Enter amount..." />
                </div>
                <button className="btn">Bid</button>
            </form>
        </div>
    )
}
     // var intAmount= Number(amount);
        // var intQuantity = Number(quantity);
        // var totalBidAmount =
        // console.log(intAmount*intQuantity);
        // console.log(amount*quantity);
   