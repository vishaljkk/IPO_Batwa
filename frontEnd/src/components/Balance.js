import React, {useContext} from 'react'
import { GlobalContext } from '../context/GlobalState';


export const Balance = () => {
    const { bankbalance } = useContext(GlobalContext);
    //console.log(bankbalance);
    return (
        <div>
            <h4>Your Account Balance</h4>
            <h1 id="balance">Rs {bankbalance}</h1>
        </div>
    )
}
