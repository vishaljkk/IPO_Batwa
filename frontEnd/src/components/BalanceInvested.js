import React, {useContext} from 'react'
import { GlobalContext } from '../context/GlobalState';

export const BalanceInvested = () => {
    const { bids, bankbalance } = useContext(GlobalContext);
  
    const amounts = bids.map(bid => bid.amount);

    const total = amounts.reduce((acc,item)=> (acc += item),0).toFixed(2);

    const remaining = bankbalance - total;
  
    const invested = Number(total);  

    return (
        <div>
            <div className="inc-exp-container">
                <div>
                    <h4>Balance</h4>
                    <p className="money plus">+Rs {remaining}</p>
                </div>
                <div>
                    <h4>Invested</h4>
                    <p className="money minus">-Rs {invested}</p>
                </div>
            </div>
        </div>
    )
}




    // const remaining = amounts
    //   .filter(item => item > 0)
    //   .reduce((acc, item) => (acc += item), 0);

    // const invested = (
    //   amounts.filter(item => item < 0).reduce((acc, item) => (acc += item), 0) *
    //   -1
    // );