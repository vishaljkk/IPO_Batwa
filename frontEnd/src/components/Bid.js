import React, {useContext} from 'react'
import { GlobalContext } from '../context/GlobalState';

export const Bid = ({bid}) => {
    const {deleteTransaction} = useContext(GlobalContext);
    const sign = bid.amount < 0 ? '-' : '+';
    return (
        <div>
            <li className={bid.amount < 0 ? 'minus' : 'plus'}>
                {bid.text} <span>{sign}Rs {Math.abs(bid.amount)}</span>
                <button className="delete-btn" onClick={()=> deleteTransaction(bid._id)}>x</button>
            </li>       
        </div>
    )
}
