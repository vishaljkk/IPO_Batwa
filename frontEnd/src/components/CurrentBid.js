import React, { useContext } from 'react';
import { GlobalContext } from '../context/GlobalState';
import { Bid } from './Bid'

export const CurrentBid = () => {
  const { bids } = useContext(GlobalContext);
  const min = 1;
  console.log(bids);
  return (
    <>
      <h3>Current Bid</h3>
      <ul id="list" className="list">
        { bids.length < min &&
          <p>No Bids Available. Lets Start Biding</p>
        }
        {bids.map(bid => (<Bid key={bid.id} bid={bid}/> ))}
        
      </ul>
    </>
  )
}

// {bids.map(bid => (<li className="minus">
//             bid.text <span>-Rs 400</span><button className="delete-btn">x</button>
//           </li>
//         ))}