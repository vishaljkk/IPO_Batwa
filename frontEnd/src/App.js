import './App.css';
import { Header } from './components/Header';
import { Balance } from './components/Balance';
import { BalanceInvested } from './components/BalanceInvested';
import { CurrentBid } from './components/CurrentBid';
import { AddBid } from './components/AddBid';
import { GlobalProvider } from './context/GlobalState';

function App() {
  return (
    <GlobalProvider>
      <div className="App">
        <Header/>
        <Balance />
        <BalanceInvested/>
        <CurrentBid/>
        <AddBid/>
      </div>
    </GlobalProvider>
  );
}

export default App;
