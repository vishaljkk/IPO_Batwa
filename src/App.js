import './App.css';
import { Header } from './components/Header';
import { Balance } from './components/Balance';
import { BalanceInvested } from './components/BalanceInvested';
import { CurrentBid } from './components/CurrentBid';
import { AddBid } from './components/AddBid';

function App() {
  return (
    <div className="App">
      <Header/>
      <Balance />
      <BalanceInvested/>
      <CurrentBid/>
      <AddBid/>
    </div>
  );
}

export default App;
