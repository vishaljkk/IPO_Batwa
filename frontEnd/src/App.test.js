import React  from 'react';
import App from './App';

import { shallow } from 'enzyme';

import { Header } from './components/Header';
import { Balance } from './components/Balance';
import { BalanceInvested } from './components/BalanceInvested';
import { CurrentBid } from './components/CurrentBid';
import { AddBid } from './components/AddBid';

import GlobalContext, { useMyContext } from "./TestContext";
import * as MyContextModule from "./TestContext";
import toJson from "enzyme-to-json";

describe('snapshots', () => {
    it('Balance snapshot', () => {
        const wrapper = shallow(<Balance />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });        
    it('App snapshot', () => {
        const wrapper = shallow(<App />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });        
    it('BalanceInvested snapshot', () => {
        const wrapper = shallow(<BalanceInvested />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });        
    it('Current Bids list snapshot', () => {
        const wrapper = shallow(<CurrentBid />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });        
    it('Add Bid snapshot', () => {
        const wrapper = shallow(<AddBid />)
        expect(toJson(wrapper)).toMatchSnapshot();
    });        
});

describe('rendering components', () => {        
    it('renders Balance component without crashing', () => {
        jest.spyOn(MyContextModule, "useMyContext").mockImplementation(() => ({
            bankbalance: 1000000
        }));
        const wrapper = shallow( <GlobalContext.Provider>
            <Balance />
          </GlobalContext.Provider>).dive();
        expect(wrapper.text()).toEqual("Your Account BalanceRs 100000");
    });
    it('renders Header component without crashing', () => {
        const wrapper = shallow(<Header />);
        const header = (<div><h2>IPO Batwa</h2></div>)
        expect(wrapper.contains(header)).toEqual(true);
    });
    it('renders Balance component with provider', () => {
        const wrapper = mount(
            <UserContext.Provider 
            value={{
              dispatch: jest.fn(),
              user: { firstname: "Alice", lastname: "Middleman" }
              loading: false
            }}>
              <Greeting />
            </UserContext.Provider>);
          expect(wrapper.find("h1").text()).toEqual("Hello Alice Middleman");
    });

});
