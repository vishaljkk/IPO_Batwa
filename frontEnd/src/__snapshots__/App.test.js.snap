// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering components renders correctly enzyme 1`] = `
<div>
  <h4>
    Your Account Balance
  </h4>
  <h1
    id="balance"
  >
    Rs 
    100000
  </h1>
</div>
`;

exports[`snapshots Add Bid snapshot 1`] = `
<div>
  <h3>
    Add Bid
  </h3>
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-control"
    >
      <label
        htmlFor="text"
      >
        Select IPO name
      </label>
      <input
        onChange={[Function]}
        placeholder="Enter IPO name..."
        type="text"
        value=""
      />
    </div>
    <div
      className="form-control"
    >
      <label
        htmlFor="amount"
      >
        Enter Number of Lots 
        <br />
      </label>
      <input
        onChange={[Function]}
        placeholder="Enter number of lots..."
        type="number"
        value={1}
      />
      <label
        htmlFor="amount"
      >
        Price Of one lot 
        <br />
      </label>
      <input
        onChange={[Function]}
        placeholder="Enter amount..."
        type="number"
        value=""
      />
    </div>
    <button
      className="btn"
    >
      Bid
    </button>
  </form>
</div>
`;

exports[`snapshots App snapshot 1`] = `
<GlobalProvider>
  <div
    className="App"
  >
    <Header />
    <Balance />
    <BalanceInvested />
    <CurrentBid />
    <AddBid />
  </div>
</GlobalProvider>
`;

exports[`snapshots Balance snapshot 1`] = `
<div>
  <h4>
    Your Account Balance
  </h4>
  <h1
    id="balance"
  >
    Rs 
    100000
  </h1>
</div>
`;

exports[`snapshots BalanceInvested snapshot 1`] = `
<div>
  <div
    className="inc-exp-container"
  >
    <div>
      <h4>
        Balance
      </h4>
      <p
        className="money plus"
      >
        +Rs 
        100000
      </p>
    </div>
    <div>
      <h4>
        Invested
      </h4>
      <p
        className="money minus"
      >
        -Rs 
        0
      </p>
    </div>
  </div>
</div>
`;

exports[`snapshots Current Bids list snapshot 1`] = `
<Fragment>
  <h3>
    Current Bid
  </h3>
  <ul
    className="list"
    id="list"
  >
    <p>
      No Bids Available. Lets Start Biding
    </p>
  </ul>
</Fragment>
`;
